src/analyze/analyze-chid.c

```diff
@@ -223,6 +252,65 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(sd_device *drm_card, char16_t **ret_panel) {
+        assert(drm_card);
+        assert(ret_panel);
+        int r;
+        const char *edid_content = NULL;
+        size_t edid_size = 0;
+        r = sd_device_get_sysattr_value_full(drm_card, "edid", &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
+
+        *ret_panel = new0(char16_t, 8);
+        if (!edid_get_panel_id(&header, *ret_panel))
+                return -EINVAL;
+        return 0;
+}
+
+static int edid_acquire(char16_t **ret_panel) {
```

Comment:
maybe call this "search" rather than "acquire" to emphasize that we are trying a bunch of things here.